### PR TITLE
Add promtheus metrics for better insight into kube-machine

### PR DESCRIPTION
**Pull request by metalmatze**
I inlined some global `var`s, I hope that's fine. Otherwise it's pretty straight forward.
closes #7

### CODE REVIEW

**Review by guusvw — APPROVED** [review]

File: cmd/controller/main.go [hunk 1]
```diff
@@ -12,39 +13,59 @@ import (
...
+		Subsystem: "controller",
+		Name:      "sync_errors_total",
+		Help:      "Total number of errors during during sync",
+	})
+	syncSeconds := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "kubemachine",
+		Subsystem: "controller",
+		Name:      "sync_seconds",
+		Help:      "Total time spend syncing in a phase in seconds",
+	}, []string{"phase"})
```
Comment (mrIncompetent) [inline cmd/controller/main.go]: Please move the metrics to a separate package

File: pkg/controller/node/controller.go [hunk 2]
```diff
@@ -65,6 +72,9 @@ func New(
 		nodeClassStore:    nodeClassStore,
 		mapi:              mapi,
 		client:            client,
+		nodes:             nodes,
+		syncErrors:        syncErrors,
+		syncSeconds:       syncSeconds,
```
Comment (mrIncompetent) [inline pkg/controller/node/controller.go]: For simplicity i would rather put them into a separate package with exported functions to interact

File: pkg/controller/node/controller.go [hunk 3]
```diff
@@ -80,6 +90,11 @@ func (c *Controller) processNextItem() bool {
 	// Invoke the method containing the business logic
 	err := c.syncNode(key.(string))
 	// Handle the error if something went wrong during the execution of the business logic
+
+	if err != nil {
+		c.syncErrors.Inc()
+	}
```
Comment (mrIncompetent) [inline pkg/controller/node/controller.go]: Please put this into `c.handleErr`

**Review by mrIncompetent — CHANGES_REQUESTED** [review]

**Comment by metalmatze** [comment]
There's no real benefit of putting this into a separate package. I'd leave it there. More packages only create more circle dependencies in the future.

**Comment by mrIncompetent** [comment]
@metalmatze It increases readability & it separates the concerns.

More packages do not create circle dependencies. Bad coding style leads to circle dependencies.

**Comment by metalmatze** [comment]
Closed & deleted in favor of #19

**Pull request closed without merging**